### PR TITLE
[Hub Generated] Review request for Microsoft.StorageCache to add version stable/2021-05-01

### PR DESCRIPTION
**Pull request by rebecca337**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/rebecca337/azure-rest-api-specs/dev-storagecache-Microsoft.StorageCache-2021-05-01...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-05-01/storagecache.json
Comment (omzevall) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-05-01/storagecache.json:1579]: This implies create-only of just the userAssignedIdentities block. Does this really belong at a higher level (CacheIdentity) because all cache identity fields are create-only?

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-05-01/storagecache.json
Comment (omzevall) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-05-01/storagecache.json:1635]: Confirmation of this behavior is still pending. Please don't merge until the _expected_ behavior is confirmed/finalized.

Comment (omzevall) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-05-01/storagecache.json:1635]: Internal Bug: 10014874

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-05-01/storagecache.json
Comment (rebecca337) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-05-01/storagecache.json:1579]: Making the appropriate change.

**Comment by rebecca337** [comment]
The LintDiff error is due to the userAssigned identity properties we added. We followed the convention that was used by all other teams in their RP that have userAssigned identities in their swagger file. Please let us know if you have any suggestions.

The cross-version breaking changes error due to read-only is NOT a change in the RP. Readonly was added because it is and always has been readonly, thus there is no change, it just didn't have the tag beforehand.

**Comment by brpanask** [comment]
> Hi, @rebecca337 your PR are labelled with WaitForARMFeedback. A notification email will be sent out shortly afterwards to notify ARM review board([armapireview@microsoft.com](mailto:armapireview@microsoft.com)). cc @

This is ready for review, its just in draft so that it is not merged until the RP rollout and ARM Manifest rollouts are completed.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-05-01/storagecache.json
Comment (omzevall) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-05-01/storagecache.json:1635]: @rebecca337 Looks like this needs to be mutable per latest PM guidance last Friday (5/28). Please remove the "create-only" tags.

Comment (rebecca337) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-05-01/storagecache.json]: Removed the tag!

**Comment by chiragg4u** [comment]
Please ensure you've reviewed following guidelines including ARM resource provider contract and REST guidelines. Estimated time (4 hours). This is required before you can request review from ARM API Review board.
 This is required before the review can be done.

**Comment by chiragg4u** [comment]
The changes looks good; once you complete the contribution guideline, I'll approve the PR from ARM side.

**Comment by leni-msft** [comment]
@msyyc @ArcturusZhang please help check the SDK breaking changes tracking errors.

**Comment by msyyc** [comment]
For python, the breaking change is expected.

**Review by JeffreyRichter — APPROVED** [review]

**Comment by brpanask** [comment]
> The changes looks good; once you complete the contribution guideline, I'll approve the PR from ARM side.

I have the approvals on the two breaking change checks and this work item to avoid a false positive on the lint diff.
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1342684

I think that should be everything needed for ARM approval.

Thanks,
Brian

File: package-lock.json
Comment (jianyexi) [inline package-lock.json:3770]: please revert this file.

Comment (rebecca337) [reply package-lock.json]: @jianyexi I have reverted the file.

**Comment by rebecca337** [comment]
@chiragg4u The breaking changes have been approved, I believe all is ready for approval from ARM side.

File: package-lock.json
Comment (rebecca337) [reply package-lock.json]: @jianyexi Changes requested have been made, could you look over it, please?

**Comment by jianyexi** [comment]
> @chiragg4u Chirag Gupta FTE The breaking changes have been approved, I believe all is ready for approval from ARM side.

still wait for the ARM feedback

**Comment by jianyexi** [comment]
@chiragg4u could you review it ?

**Comment by rebecca337** [comment]
@jianyexi PR has been approved from ARM side and is ready to merge.

**Comment by brpanask** [comment]
> /azp run

 @jianyexi Can you merge this?  We need to start generating the sdks and docs soon and need to get this merged.
Thanks.

**Pull request merged**
